subconscious/model.py

```diff
@@ -24,23 +42,27 @@ def __init__(self, type=str, primary_key=None, composite_key=None, index=None,
         You can't have both a primary_key and composite_key in the same model.
         index is whether you want this column indexed or not for faster retrieval.
         """
-        # TODO: support for other field types (datetime, etc)
+        if type not in (str, int):
+            # TODO: support for other field types (datetime, uuid, etc)
+            err_msg = 'Bad Field Type: {}'.format(type)
+            raise InvalidColumnDefinition(err_msg)
+
+        if primary_key and composite_key:
```

Comment:
That's fair!

I think how we got to this API design was that I wanted users to be super explicit with their model design. If someone was setting `primary_key=True` on multiple columns that felt incorrect.
I just added some (limited) test coverage for composite keys, it's easier to talk about when you have an example to look at. It turns out, they didn't behave quite as I had thought!

https://github.com/paxos-bankchain/subconscious/pull/10/commits/201daeee1dc6c156bc3db3c91288cc17ebc1f96b#diff-248503ab49834a9276ab5a3556b8d73dR39

You can have `index=True` on multiple columns, but we do require one to be the `primary_key`. Perhaps that restriction could be relaxed? Typically, I think one feature of composite keys is that they guarantee uniqueness across the compound key.

I'm not sure what the best approach is here, def open to improvement though.